Plugins/NativeEngine/Source/ShaderCompilerOpenGL.cpp

```diff
@@ -9,151 +10,48 @@
 namespace Babylon
 {
     extern const TBuiltInResource DefaultTBuiltInResource;
-
+    
     namespace
     {
         void AddShader(glslang::TProgram& program, glslang::TShader& shader, std::string_view source)
         {
             const std::array<const char*, 1> sources{source.data()};
             shader.setStrings(sources.data(), gsl::narrow_cast<int>(sources.size()));
-            shader.setEnvInput(glslang::EShSourceGlsl, shader.getStage(), glslang::EShClientVulkan, 100);
-            shader.setEnvClient(glslang::EShClientVulkan, glslang::EShTargetVulkan_1_0);
-            shader.setEnvTarget(glslang::EShTargetSpv, glslang::EShTargetSpv_1_0);
 
-            if (!shader.parse(&DefaultTBuiltInResource, 450, false, EShMsgDefault))
+            if (!shader.parse(&DefaultTBuiltInResource, 310, EProfile::EEsProfile, true, true, EShMsgDefault))
             {
-                throw std::exception();
+                throw std::runtime_error(shader.getInfoDebugLog());
             }
 
             program.addShader(&shader);
         }
 
-        std::unique_ptr<spirv_cross::Compiler> CompileShader(glslang::TProgram& program, EShLanguage stage, std::string& glsl)
+        std::pair<std::unique_ptr<spirv_cross::Parser>, std::unique_ptr<spirv_cross::Compiler>> CompileShader(glslang::TProgram& program, EShLanguage stage, std::string& glsl)
         {
             std::vector<uint32_t> spirv;
             glslang::GlslangToSpv(*program.getIntermediate(stage), spirv);
```

Comment:
Do we still need to go to SPIRV for OpenGL?

Reply:
Right now, the logic in `NativeEngine` still uses the SPIRV types to build the bgfx representation of the shader. Ultimately that logic should probably be moved out of `NativeEngine` and adapted to use glslang features instead of SPIRV. However, that looks like a fairly substantial change with only a tangential connection to this one, so my opinion is that we should leave that to its own PR. I've made an issue to track this: #300.